chore: Add setRequiredFields and setIgnoredFields documentation (#911)
SAP/cloud-sdk: docs-js/features/odata/common/operations/update.mdx

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> The Update request builder allows you to change existing entities.

Reviewer: ```suggestion
The update request builder allows you to change existing entities.
```

By default, `PATCH` is used to only update changed fields.

In the following example, we first get a business partner, change one of its values, and then send an update request.

<Tabs groupId="version" defaultValue="major" values={[ { label: 'SDK 2.x', value: 'major' }, { label: 'SDK 1.x', value: 'deprecated' } ]}>
<TabItem value="major">

```ts
const { businessPartnerApi } = businessPartnerService();

> // First we get a businesspartner with the id of "1".

Reviewer: ```suggestion
// Get a business partner
```

const businessPartner = businessPartnerApi
    .requestBuilder()
    .getByKey('1')
    .execute({destinationName: 'myDestination'});

> // Then we change the business partner's first name to "Steve".

Reviewer: ```suggestion
// Change first name
```

businessPartner.then((businessPartnerResponse) => (businessPartnerResponse.firstName = 'Steve'));

> // Finally we update the business partner with the updated first name.

Reviewer: ```suggestion
// Send a PATCH request with `{ "FirstName" : "Steve" }`
```

businessPartnerApi
    .requestBuilder()
    .update(businessPartner);
```

</TabItem>
<TabItem value="deprecated">

```ts

> // First we get a businesspartner with the id of "1".

Reviewer: ```suggestion
// Get a business partner
```

const businessPartner = BusinessPartner
    .requestBuilder()
    .getByKey('1')
    .execute({destinationName: 'myDestination'});

> // Then we change the business partner's first name to "Steve".

Reviewer: ```suggestion
// Change first name
```

businessPartner.then((businessPartnerResponse) => (businessPartnerResponse.firstName = 'Steve'));

> // Finally we update the business partner with the updated first name.

Reviewer: ```suggestion
// Send a PATCH request with `{ "FirstName" : "Steve" }`
```

BusinessPartner
    .requestBuilder()
    .update(businessPartner);
```

</TabItem>
</Tabs>

> In the example above only the changed field of the given `businessPartner` instance,
> so in this case the first name, is sent to the `BusinessPartner` service using `PATCH`.

Reviewer: This sentence reads weird, but I am not so happy with my suggestion either. Maybe it is now duplicate with the example?

```suggestion
In the example above we changed the first name of the given business partner.
The payload sent to the service with PATCH includes only the first name.
```

> Be aware that update requests will fail if their ETag doesn't match.

Reviewer: ```suggestion
Be aware that update requests will fail if their ETags don't match.
```

In the example above only the changed fields of the given `businessPartner` instance are sent to the `BusinessPartner` service using `PATCH`.

Be aware that update requests will fail if their ETag doesn't match.

> Check out our [ETag](#handling-of-etags) section for more information.

Reviewer: [nth] In the text above we say that PUT + PATCH can be used and that "only changed" fields will be updated. I think it would be nice to improve the description as follows:

- State that the update request builder allows to change existing entities. Do not mention PUT and PATCH for this. We already state that PATCH is used by default for the changed fields.
- To understand what changed fields are we should explain it somehow. Maybe we can add this to the example and first do a GET and change one property and then do the update. With this it will become much more clear that the data that is sent should only contain the changed property. (also state it in the text). This is a nice basis for the PUT section and also the setIgnored/Required fields.
